tests/test_core.py

```diff
@@ -7154,6 +7154,47 @@ def test_async_exit(self):
 }
 ''', 'f\nhello\nf\nhello\nf\nhello\nf\nhello\nf\nhello\nexit\n')
 
+  def test_async_abort(self):
+    if not self.is_emterpreter(): return self.skip('emterpreter-only test')
+
+    self.banned_js_engines = [SPIDERMONKEY_ENGINE, V8_ENGINE] # needs setTimeout which only node has
+
+    Settings.EMTERPRETIFY_ASYNC = 1
+
+    open('lib.js', 'w').write(r'''
+mergeInto(LibraryManager.library, {
+  sleep_with_abort__deps: ['$EmterpreterAsync'],
+  sleep_with_abort: function() {
+    EmterpreterAsync.handle(function(resume) {
+      setTimeout(function() {
```

Comment:
I'm not sure the spec mandates that the 50ms callback happen before the 100ms. Also, there's some risk of a JS engine bug reordering them.

How about calling the second setTimeout inside the callback for the first, which has guaranteed ordering?

Reply:
Done.